### ISSUE TITLE
Parsing a yaml to an object without scalar value

### ISSUE THREAD

**Issue by Will3333** [charleskorn/kaml#6]
Hello,

First of all, thank you for your work.

I have serialized an object of the class Conf below to a yaml file.

```kotlin
@Serializable
data class Conf(val api : API = API(), val database : DataBase)

@Serializable
data class API(val port : Int = 9000)

@Serializable
data class DataBase(val filesStorage : FilesStorage? = null, val server : Server? = null)

@Serializable
data class FilesStorage(val system : String = "HSQLDB")

@Serializable
data class Server(val postgresql : Postgresql? = null, val mariaDb : MariaDb? = null)

@Serializable
data class Postgresql(val host : String, val port : Int = 5432, val user : String, val password : String, val dbName : String)

@Serializable
data class MariaDb(val host : String, val port : Int = 3306, val user : String, val password : String, val dbName : String)
```
```kotlin
val config = Conf.assistUserToGetInstance()
val configYaml = Yaml(configuration = YamlConfiguration(encodeDefaults = false)).stringify(Conf.serializer(), config)
configFile.writeText(configYaml)
```

The produced yaml :
```yaml
database:
  server:
    mariaDb:
      host: "db.test.com"
      user: "toto"
      password: "fzrefgl"
      dbName: "toto"
```

However I failed to parse this same yaml.

```kotlin
val configRawYaml = configFile.readText()
Yaml.default.parse(Conf.serializer(), configRawYaml)
```
```
Exception in thread "main" com.charleskorn.kaml.InvalidPropertyValueException: Value for 'server' is invalid: Expected a (possibly null) scalar value, but got a map
	at com.charleskorn.kaml.YamlMapInput.fromCurrentValue(YamlInput.kt:246)
```

Do you have a suggestion ?
I don't understand why it wait for a scalar value.

**Comment by charleskorn** [comment]
Hi @Will3333, thanks for the issue report. Looks like there's a bug in how we're handling nullable values in objects. I'll take a look in the next few days and post an update here once I've fixed the issue.

**Comment by charleskorn** [comment]
I've just fixed this and this will be part of v0.13, which should be available in the next hour or so.

**Comment by Will3333** [comment]
Thank you.
